analysis/study_definition.py

```diff
@@ -1,13 +1,884 @@
+  myasthenia_gravis_nhsd_icd10=comorbidity_icd(myasthenia_gravis_nhsd_icd10_codes),
+  huntingtons_disease_nhsd_snomed=comorbidity_snomed(huntingtons_disease_nhsd_snomed_codes),
+  huntingtons_disease_nhsd_icd10=comorbidity_icd(huntingtons_disease_nhsd_icd10_codes),
+  huntingtons_disease_nhsd = patients.maximum_of("huntingtons_disease_nhsd_snomed", "huntingtons_disease_nhsd_icd10"),
+  rare_neuro = patients.maximum_of("multiple_sclerosis_nhsd_snomed", "multiple_sclerosis_nhsd_icd10", 
+                                                  "motor_neurone_disease_nhsd_snomed", "motor_neurone_disease_nhsd_icd10", 
+                                                  "myasthenia_gravis_nhsd_snomed", "myasthenia_gravis_nhsd_icd10",
+                                                  "huntingtons_disease_nhsd_snomed", "huntingtons_disease_nhsd_icd10"),
+
+  ## Eligible // need to reduce eligibility of imids to those withh imid on drug/HCD
+  eligible = patients.maximum_of("downs_syndrome", 
+                                "solid_cancer", 
+                                "haem_disease",
+                                "renal_disease",
+                                "liver_disease",
+                                "immunosupression",
+                                "imid_on_drug", 
+                                "hiv_aids",
+                                "solid_organ_transplant",
+                                "rare_neuro"),
+  
+  ## Blueteq ‘high risk’ cohort
+  high_risk_cohort_covid_therapeutics = patients.with_covid_therapeutics(
+    with_these_therapeutics = ["Sotrovimab", "Paxlovid", "Molnupiravir"],
+    with_these_indications = "non_hospitalised",
+    on_or_after = "index_date",
+    find_first_match_in_period = True,
+    returning = "risk_group",
+    date_format = "YYYY-MM-DD",
+    return_expectations = {
+      "rate": "universal",
+      "incidence": 0.4,
+      "category": {
+        "ratios": {
+          "Downs syndrome": 0.1,
+          "sickle cell disease": 0.1,
+          "solid cancer": 0.1,
+          "haematological diseases": 0.05,
+          "stem cell transplant recipients": 0.05,
+          "renal disease": 0.1,
+          "liver disease": 0.05,
+          "IMID": 0.1,
+          "IMID,solid cancer": 0.1,
+          "haematological malignancies": 0.05,
+          "primary immune deficiencies": 0.1,
+          "HIV or AIDS": 0.05,
+          "NA":0.05,},},
+    },
+  ),
+  
+  eligible_binary= patients.satisfying(
+    """
+    eligible 
+    OR
+    high_risk_cohort_covid_therapeutics
+    """,
+    return_expectations={
+       "incidence": 0.5,
     },
-    population=patients.registered_with_one_practice_between(
-        "2019-02-01", "2020-02-01"
+  ),
+  
+  ## TREATMENT - MAB + Antivirals. 
+  sotrovimab=covid_therapeutics("Sotrovimab"),
+  molnupiravir =covid_therapeutics("Molnupiravir"),
+  paxlovid=covid_therapeutics("Paxlovid"),
+  remdesivir=covid_therapeutics("Remdesivir"),
```

Comment:
Necessary? Please explain.

Reply:
I have included remdesivir and casirivimab because if they have had either of these drugs, after their positive covid test then they are excluded from the treatment groups but also excluded from the control group

Reply:
also in regard to the “statuses” field in the therapeutics data, I used this as Bang had for the efficacy. He had used similar code in his study definition. 

I have now only keep the with_these_statuses = ["Treatment Not Started"] for my study definition as I then exclude anyone with a code ["Treatment Not Started"] as I don’t want them to be included in the safety analysis of drug if they didn’t receive any drug. What are you thought?

Reply:
If you're not interested in patients who get remdesivir or casirivimab or with_these_statuses = ["Treatment Not Started"] then best to exclude patients at the begging in the population = patients.satisfying section. The only reason to keep them would be if you wanted to know the number excluded by this filter (i.e. for a flow chart of inclusion/because it's meaningful to know) or for some sort of sensitivity analysis

Reply:
Hmm, i think i would like to know the numbers of "Treatment Not Started"] , so i can state how many actually had the drug, and likewise with remdesivir or casirivimab